docs/source/udf/profiling.rst

```diff
@@ -0,0 +1,180 @@
+.. _`udf profiling`:
+
+.. testsetup:: *
+
+    from libertem.api import Context
+
+
+Profiling UDFs
+==============
+
+To bring your UDF to production, it may be necessary to look into the performance and
+efficiency of your function. There may be some low-hanging fruit to optimize away, but
+they only really become visible once you profile your code. Just looking at code and guessing
+which parts are expensive can be misleading and cause you to spend time optimizing the wrong part
+of your code. Always measure!
+
+Profiling means instrumenting the execution of a program and finding out which parts
+of your code use how much of a given resource (for example, CPU time, or memory). 
+
+Prerequisite: inline executor
+-----------------------------
+
+By default, your UDF will be run in a multi-process or multi-threaded environment. This makes
+profiling a challenge. But as most of the time, improving the execution time of your UDF in a
```

Comment:
...since the usual profiling tools will only capture information on the main process that performs the final reduction and not the processes that do most of the work.